### PR TITLE
Fixes Triple Dive effect

### LINKED ISSUE #2795: Triple Dive uses EFFECT_TRIPLE_KICK when it shouldn't
According to Bulbapedia and Smogon, Triple Dive's only effect is that it hits 3 times, and each consecutive hit shouldn't increase in damage like Triple Kick/Axel do.

image
image

### PR DESCRIPTION
**Pull request by AlexOn1ine**
## Description
Fixes Triple Dive effect by adding `FLAG_THREE_STRIKES`. The flag is additionally used for Surging Strikes. 

## Images
Triple Dive:
triple_dive
Surging Strikes:
surging_strikes

## Issue(s) that this PR fixes
Fixes #2795

### CODE REVIEW

File: src/battle_util.c [hunk 1]
```diff
@@ -3739,7 +3739,7 @@ u8 AtkCanceller_UnableToUseMove(void)
                     // TODO
                 }
             }
-            else if (gBattleMoves[gCurrentMove].effect == EFFECT_TRIPLE_KICK || gCurrentMove == MOVE_SURGING_STRIKES)
+            else if (gBattleMoves[gCurrentMove].flags & FLAG_THREE_STRIKES)
```
Comment (AsparagusEduardo) [inline src/battle_util.c:3742]: I think this breaks Triple Kick/Axel.

Comment (AlexOn1ine) [reply src/battle_util.c]: Fixed + added a couple tests

File: test/move_flag_three_strikes.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TRIPLE_DIVE, player);
+        HP_BAR(opponent, captureDamage: &thirdHit);
+    } FINALLY {
+        EXPECT_EQ(firstHit, secondHit);
+        EXPECT_EQ(secondHit, thirdHit);
+        EXPECT_EQ(firstHit, thirdHit);
+    }
+}
+
+SINGLE_BATTLE_TEST("Three Strikes on Surging Strikes do critical damage")
```
Comment (AsparagusEduardo) [inline test/move_flag_three_strikes.c:30]:
```suggestion
SINGLE_BATTLE_TEST("Surging Strikes hits 3 times with each hit being a critical hit")
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Comment by AlexOn1ine** [comment]
Applied suggestions.

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**